README.md

```diff
@@ -1,6 +1,112 @@
+
+   ```csharp
+        if (fieldValue == null)
+        {
+            record.Add(field.Name, null);
+            return;
+        }
+
+        fieldSchema = GetNonNullablePartOfNullableSchema(field.Schema);
+        fieldType = fieldSchema.Tag;
+
+        ...
+
+        private Schema GetNonNullablePartOfNullableSchema(Schema nullableSchema)
+        {
+            var jsonSchema = (JsonElement) JsonSerializer.Deserialize<object>(nullableSchema.ToString());
+            jsonSchema.TryGetProperty("type", out var unionList);
+            var notNullSchema = unionList.EnumerateArray().First(type => type.ToString() != "null").ToString();
+            return Schema.Parse(notNullSchema);
+        }
+   ```
+
+   **Enums**:
+
+   ```csharp 
+        new GenericEnum((EnumSchema) fieldSchema, fieldValue.ToString())
+   ```
+
+   **DateTime Objects**:
+
+   Converts to a int holding a Unix timestamp.
+
+    ```csharp
+        private int? UnixTimestampNullable(object obj)
+        {
+            var date = (DateTime?) obj;
+            return (int?) (date?.Subtract(new DateTime(1970, 1, 1)))?.TotalSeconds;
+        }
+    ```
+
+    **Arrays**:
+
+    ```csharp
+        var fieldValueAsList = (List<ExampleList>) fieldValue;
+        var itemsSchema = GetSchemaForArrayItems(fieldSchema);
+        var recordsList = fieldValueAsList.Select(listItem => PopulateFields(listItem, itemsSchema)).ToArray()
+        record.Add(field.Name, recordsList);
+        
+        private Schema GetSchemaForArrayItems(Schema arraySchema)
+        {
+            var jsonSchema = (JsonElement) JsonSerializer.Deserialize<object>(arraySchema.ToString());
+            jsonSchema.TryGetProperty("items", out var itemsSchemaJson);
+            return Schema.Parse(itemsSchemaJson.ToString());
+        }
+    ``` 
+
+    Once the data types have been converted you can create the Generic Record by adding each value to the record, this can be done through the use of a for loop & reflection:
+
+    ```csharp
+
+        public GenericRecord PopulateFields(object item, Schema schema)
+        {
+            var record = new GenericRecord((RecordSchema) schema);
+            ((RecordSchema) schema).Fields.ForEach(field =>
+            {
+                PropertyInfo propInfo = item.GetType().GetProperty(field.Name);
+                if (propInfo == null)
+                {
+                    Console.WriteLine($"Field name: {field.Name} not found in {item}");
+                    return;
+                }
+
+                var fieldValue = propInfo.GetValue(item);
+                var fieldSchema = field.Schema;
+                var fieldType = field.Schema.Tag;
+
+                //Add any data type conversion required here
+
+                record.Add(field.Name, fieldValue);
+            });
+
+            return record;
+        }
+
+    ```
+
+6. Lastly the record created above is then sent through to Kafka.
```

Comment:
Maybe separate this into two sections with the following titles: 
Serializing the data

Sending the data to Kafka